### PR TITLE
HomePage -> CoursePage routing + CoursePage header

### PR DESCRIPTION
**Pull request by illeanz**
If you click on the course name from the homepage list, it will now take you to its respective course page! 

The course overview has its info populated based on the course. Instead of top review, course overview shows the most recent review.

### CODE REVIEW

File: react_app/src/components/HomePage/CourseItem.js [hunk 1]
```diff
@@ -41,7 +41,10 @@ class CourseItem extends React.Component {
                 <Box className='courseItemVerticalSides courseItemVerticalMiddle'
                      style={{paddingTop: '5px'}}
                 >
-                    <Typography variant='h5'>{this.props.courseNumber || 'CPSC110'} </Typography>
+                    <Link to={'/courses/' + this.props.courseNumber.toString()}
+                            style={{textDecoration: 'none', color: 'black'}}>
+                        <Typography variant='h5'>{this.props.courseNumber || 'CPSC110'} </Typography>
```
Comment (illeanz) [inline react_app/src/components/HomePage/CourseItem.js:46]: not sure if we should keep the cpsc 110 fallback here

**Comment by illeanz** [comment]
Haven't really figured out how to pull in the new changes yet tho :<

**Review by aaronNgu — APPROVED** [review]
It works!

File: react_app/src/components/CoursePage/CourseOverview.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import {Box, Typography} from '@material-ui/core';
+import Rating from './Rating';
+import RateCourseButton from './RateCourseButton';
+import {connect} from 'react-redux';
+import {fetchCourseInfo, fetchReviews, toggleAddReviewBox} from '../../actions';
+
+class CourseOverview extends React.Component {
+
+    constructor(props) {
+        super(props);
```
Comment (aaronNgu) [inline react_app/src/components/CoursePage/CourseOverview.js]: I get a warning that says Useless Contructor.
Screen Shot 2020-07-22 at 7 20 19 PM

Comment (illeanz) [reply react_app/src/components/CoursePage/CourseOverview.js]: oh yea youre right lemme get rid of that

**Comment by wwyc** [comment]
something is look strange, under Files Changed it looks like you adding changes that are already there in master.  The original files also don't look like what it is on master now?

File: package.json [hunk 3]
```diff
@@ -3,12 +3,14 @@
   "version": "0.0.0",
   "private": true,
   "scripts": {
-    "start": "node ./bin/www"
+    "start": "npm run build --prefix ./react_app && node ./bin/www",
+    "heroku-postbuild": "cd react_app && npm install --only=dev && npm install && npm run build"
```
Comment (wwyc) [inline package.json:7]: for deployment?

File: routes/reviews.js [hunk 4]
```diff
@@ -4,6 +4,21 @@ const mongoose = require('mongoose')
 const Review = require('../models/review')
 
 /* GET Reviews listing. */
+router.get('/course/:courseId', function (req, res, next) {
```
Comment (wwyc) [inline routes/reviews.js:7]: this was moved up?

**Comment by wwyc** [comment]
something is look strange, under Files Changed it looks like you are adding changes that are already there in master. The original files also don't look like what it is on master now?
I made a copy of your branch and created a [PR](https://github.com/aaronNgu/CourseHub/pull/48/files) just to see the difference.

You might want to make a copy of your branch and try creating another PR to see?

otherwise the functionality works! thx so much for doing this :)

File: routes/reviews.js (hunk 4 above)
Comment (illeanz) [reply routes/reviews.js:7]: Yeah i moved it up because it's more specific

**Comment by illeanz** [comment]
Thanks for pointing it out!! I'll check again. Sorry I've been so late with this...

**Review by Spandana-Baruah — APPROVED** [review]

**Comment by wwyc** [comment]
I just realize the api and course page uses the same url /course/:courseId, if you refresh the course page it takes you to the api route

**Comment by illeanz** [comment]
@wwyc Should I change one of them to courses/courseid?

**Pull request closed without merging**